src/meta/src/manager/catalog/mod.rs

```diff
@@ -1423,6 +1432,33 @@ impl CatalogManager {
             .map(|sink_id| sinks.remove(*sink_id).unwrap())
             .collect_vec();
 
+        if !matches!(relation, RelationIdEnum::Sink(_)) {
+            let table_sinks = sinks_removed
+                .iter()
+                .filter(|sink| {
+                    if let Some(target_table) = sink.target_table {
+                        // Table sink but associated with the table
+                        if matches!(relation, RelationIdEnum::Table(table_id) if table_id == target_table) {
+                            false
+                        } else {
+                            // Table sink
+                            true
+                        }
+                    } else {
+                        // Normal sink
+                        false
+                    }
+                })
+                .collect_vec();
+
+            if !table_sinks.is_empty() {
+                bail!(
+                    "Found {} sink(s) into table in dependency, please drop them manually",
+                    table_sinks.len()
+                );
```

Comment:
Does it mean that cascadely dropping a view with table sink dependencies will be rejected?